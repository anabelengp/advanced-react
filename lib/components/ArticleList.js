import React from 'react';

import Article from './Article';

function ArticleList(props) {
  return (
    <div>
      {Object.values(props.articles).map(article => (
        <Article key={article.id} article={article} store={props.store} />
      ))}
    </div>
  );
}

export default ArticleList;
