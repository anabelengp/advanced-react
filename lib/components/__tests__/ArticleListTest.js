import React from 'react';
import renderer from 'react-test-renderer';

import ArticleList from '../ArticleList';

describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: {
        id: 'a',
        authorId: 'au1',
      },
      b: {
        id: 'b',
        authorId: 'au2',
      },
    },
    store: { lookupAuthor: jest.fn(() => ({})) },
  };
  it('renders correctly', () => {
    const tree = renderer.create(<ArticleList {...testProps} />).toJSON();
    expect(tree.children.length).toBe(2);

    // jest snapshot testing
    expect(tree).toMatchSnapshot();
  });
});
