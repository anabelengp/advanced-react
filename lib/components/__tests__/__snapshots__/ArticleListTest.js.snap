// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div>
    <div />
    <div>
      Invalid Date
    </div>
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
  <div>
    <div />
    <div>
      Invalid Date
    </div>
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
</div>
`;
